Fix to Learnable Fake Quantization Op Benchmarking

paulshaoyuqiao opened the pull request:
Summary: In this diff, a fix is added where the original non-learnable fake quantize is provided with trainable scale and zero point, whereas the requires_grad for both parameters should be completely disabled.

Test Plan:
Use the following command to execute the benchmark test:

`buck test mode/dev-nosan pt:quantization_test`

Differential Revision: D23107846

vadimkantorov commented on benchmarks/operator_benchmark/pt/quantization_test.py:
```diff
@@ -233,6 +233,8 @@ def init(self, N, C, H, W, nbits, device, op_type):
         self.scale = torch.tensor([1.] * C).to(device)
```
is it equivalent to `torch.ones(C, device = device)`?

paulshaoyuqiao replied:
I think it would be more similar to `torch.ones(C, device=device, dtype=torch.float32)

vadimkantorov replied:
And similar case for torch.zeros as well around this line...

paulshaoyuqiao replied:
For zero point, it will need to be `torch.zeros(C, device=device, dtype=torch.int64)` instead. Fake quantization kernel expects zero point to be of type Long.

vadimkantorov replied:
and more colloquial as well

Commit:
Fix to Learnable Fake Quantization Op Benchmarking (#43018)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/43018

In this diff, a fix is added where the original non-learnable fake quantize is provided with trainable scale and zero point, whereas the requires_grad for both parameters should be completely disabled.

Test Plan:
Use the following command to execute the benchmark test:

`buck test mode/dev-nosan pt:quantization_test`

Differential Revision: D23107846

fbshipit-source-id: fae7be3f365a926d601ea99693c37edb30e0b880

vkuzo reviewed (approved):
thanks for fixing this!
